drivers/rtc/Kconfig.smartbond

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config RTC_SMARTBOND
+	bool "Smartbond RTC driver"
+	depends on DT_HAS_RENESAS_SMARTBOND_RTC_ENABLED
```

Comment:
this normally goes with a `default y` so you don't have to explicitly enable the driver once it's instantiated in the devicetree

Reply:
I did not really get that comment. Should I remove the dependency completely?

Reply:
normally drivers backed by a devicetree node use a pattern like

```
depends on DT_HAS_..._ENABLED
default y
```

that means that the option is forced to "no" if there's no node with that compatible in the tree, but that it defaults to "yes" if there's any. That's convenient as it means you don't have to manually have a `CONFIG_RTC_SMARTBOND=y` in your application, but rather just have the node enabled in the tree.

So this should have a `default y`, which means that it's enabled by default as long as there's a compatible node defined.